lib/ts-sdk/src/client/client.ts

```diff
@@ -0,0 +1,199 @@
+/**
+ * Main HTTP client for the CommonGrants API.
+ */
+
+import { type ClientConfig, type ResolvedConfig, resolveConfig } from "./config";
+import { Auth, buildAuthHeaders, type AuthMethod } from "./auth";
+import { Opportunities } from "./opportunities";
+import type { Paginated } from "@/types";
+
+// =============================================================================
+// FetchManyOptions interface
+// =============================================================================
+/** Options for the fetchMany auto-pagination method. */
+export interface FetchManyOptions {
+  /** Starting page number (default: 1) */
+  page?: number;
+  /** Items per page (uses client default if not specified) */
+  pageSize?: number;
+  /** Maximum total items to fetch (uses client default if not specified) */
+  maxItems?: number;
+  /** Abort signal for cancellation */
+  signal?: AbortSignal;
+}
+
+// =============================================================================
+// Client class
+// =============================================================================
+
+/**
+ * HTTP client for interacting with the CommonGrants API.
+ *
+ * @example
+ * ```ts
+ * import { Client, Auth } from "@common-grants/sdk/client";
+ *
+ * const client = new Client({
+ *   baseUrl: "https://api.example.org",
+ *   auth: Auth.bearer("your-token"),
+ * });
+ *
+ * // Get an opportunity
+ * const opp = await client.opportunities.get("opp-123");
+ * console.log(opp.title);
+ *
+ * // List opportunities
+ * const list = await client.opportunities.list({ page: 1 });
+ * ```
+ */
+export class Client {
+  private readonly config: ResolvedConfig;
+  private readonly auth: AuthMethod;
+
+  /** Opportunities resource namespace */
+  public readonly opportunities: Opportunities;
+
+  // =============================================================================
+  // Client constructor
+  // =============================================================================
+
+  constructor(options: ClientConfig & { auth?: AuthMethod }) {
+    this.config = resolveConfig(options);
+    this.auth = options.auth ?? Auth.none();
+
+    // Initialize resource namespaces
+    this.opportunities = new Opportunities(this);
+  }
+
+  // =============================================================================
+  // Client.fetch
+  // =============================================================================
+
+  /**
+   * Makes an authenticated fetch request to the API.
+   *
+   * @param path - API path (will be appended to baseUrl)
+   * @param init - Fetch init options
+   * @returns Fetch Response
+   *
+   * @example
+   * ```ts
+   * const response = await client.fetch("/common-grants/opportunities");
+   * const data = await response.json();
+   * ```
+   */
+  async fetch(path: string, init?: RequestInit): Promise<Response> {
```

Comment:
This implicitly supports other methods besides `GET` with the optional `init?: RequestInit` param. But I plan to make this more ergonomic in #429